File: Java/src/GUI/Box.java
package GUI;

import javax.swing.*;
import java.awt.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.awt.event.KeyEvent;
import java.awt.event.KeyListener;

import World.World;
import World.Point;
import World.body.animal.*;
import World.body.plant.*;

public class Box extends JButton{
    private World world;
    private int x;
    private int y;
    private Color color = Color_obj.EMPTY.getColor();

    public Box(int x, int y, World world) {
        this.world = world;
        this.x = x;
        this.y = y;
        this.setBackground(color);
        setFocusable(false);
        this.addActionListener(new ActionListener() {
            @Override
            public void actionPerformed(ActionEvent e) {
                if(color.equals(Color_obj.EMPTY.getColor())) {
                    String[] possibleValues = {"Antelope", "Fox", "Sheep", "Tortoise", "Wolf", "Belladonna", "Dandelion", "Grass", "Guarana", "Sosnowsky's Hogweed"};

                    Object selectedValue = JOptionPane.showInputDialog(null,
                            "Choose one", "Choose new body",
                            JOptionPane.INFORMATION_MESSAGE, null,
                            possibleValues, possibleValues[0]);
                    if (selectedValue != null) {
                        if (selectedValue.equals("Antelope")) {
                            world.add_body(new Antelope(new Point(x, y), world));
                            world.getNews_panel().add_news("New Antelope at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Fox")) {
                            world.add_body(new Fox(new Point(x,y), world));
                            world.getNews_panel().add_news("New fox at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Sheep")) {
                            world.add_body(new Sheep(new Point(x, y), world));
                            world.getNews_panel().add_news("New sheep at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Tortoise")) {
                            world.add_body(new Tortoise(new Point(x, y), world));
                            world.getNews_panel().add_news("New tortoise at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Wolf")) {
                            world.add_body(new Wolf(new Point(x, y), world));
                            world.getNews_panel().add_news("New wolf at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Belladonna")) {
                            world.add_body(new Belladonna(new Point(x, y), world));
                            world.getNews_panel().add_news("New belladonna at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Dandelion")) {
                            world.add_body(new Dandelion(new Point(x, y), world));
                            world.getNews_panel().add_news("New dandelion at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Grass")) {
                            world.add_body(new Grass(new Point(x, y), world));
                            world.getNews_panel().add_news("New grass at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Guarana")) {
                            world.add_body(new Guarana(new Point(x, y), world));
                            world.getNews_panel().add_news("New guarana at (" + x + "," + y + ")");
                        } else if (selectedValue.equals("Sosnowsky's Hogweed")) {
                            world.add_body(new SosnowskysHogweed(new Point(x, y), world));
                            world.getNews_panel().add_news("New Sosnowsky's Hogweed at (" + x + "," + y + ")");
                        } else {
                            world.getNews_panel().add_news("Are (" + x + "," + y + ") is occupied!");
                        }
                    }
                }

            }
        });
    }

    public void setColor(Color_obj color2){
        color=color2.getColor();
        this.setBackground(color);
    }

    public Color getColor() {
        return color;
    }

}

File: Python/GUI/Box.py
import tkinter as tk
from tkinter import simpledialog, messagebox

from GUI.colors import Colors
from World.Body.Animal.antelope import Antelope
from World.Body.Animal.cybersheep import CyberSheep
from World.Body.Animal.fox import Fox
from World.Body.Animal.sheep import Sheep
from World.Body.Animal.tortoise import Tortoise
from World.Body.Animal.wolf import Wolf
from World.Body.Plant.dandelion import Dandelion
from World.Body.Plant.belladonna import  Belladonna
from World.Body.Plant.grass import Grass
from World.Body.Plant.guarana import Guarana
from World.Body.Plant.sosnowskys_hogweed import SosnowskysHogweed
from World.point import Point


class Box:
    world = None
    board = None
    x = None
    y = None
    color = Colors.EMPTY
    button = None
    option_var = None

    def __init__(self, world, board, x, y):
        self.option_var = tk.StringVar()
        self.world = world
        self.board = board
        self.x = x
        self.y = y
        self.pixelVirtual = tk.PhotoImage(width=1, height=1)
        self.button = tk.Button(board, image=self.pixelVirtual, width=23, height=23, command=self.handle_click)
        self.button.config(bg=self.color.value)

    @property
    def get_button(self):
        return self.button

    def get_color(self):
        return self.color

    def set_color(self, color):
        self.color = color
        self.button.config(bg=self.color.value)

    def handle_click(self):
        if self.color == Colors.EMPTY:
            # print(f"Kliknięto przycisk na pozycji ({self.x}, {self.y})")
            choice_window = tk.Toplevel(self.board)
            options = ['Antelope', 'Cyber sheep', 'Fox',  'Sheep', 'Tortoise', 'Wolf', 'Grass', 'Guarana','Sosnowsky\'s hogweed', 'Dandelion', 'Belldonna']

            self.option_var.set(options[0])

            for option in options:
                rb = tk.Radiobutton(choice_window, text=option, variable=self.option_var, value=option)
                rb.pack(anchor='w')
            confirm_button = tk.Button(choice_window, text="Submit", command=self.add_body)
            confirm_button.pack()

    def add_body(self):
        if self.option_var.get() == "Antelope":
            self.world.add_body(Antelope(Point(self.y+1, self.x+1),self.world))
        elif self.option_var.get() == "Cyber sheep":
            self.world.add_body(CyberSheep(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Fox":
            self.world.add_body(Fox(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Sheep":
            self.world.add_body(Sheep(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Tortoise":
            self.world.add_body(Tortoise(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Wolf":
            self.world.add_body(Wolf(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Grass":
            self.world.add_body(Grass(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Guarana":
            self.world.add_body(Guarana(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Sosnowsky\'s hogweed":
            self.world.add_body(SosnowskysHogweed(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Dandelion":
            self.world.add_body(Dandelion(Point(self.y+1, self.x+1), self.world))
        elif self.option_var.get() == "Belldonna":
            self.world.add_body(Belladonna(Point(self.y+1, self.x+1), self.world))


